File: tests/fixtures/edgecase_repo/php/module2.php
<?php
namespace Example\Gen;

use Example\Gen\Base2;
use Example\Gen\Handler2 as H2;

class Base2
{
    protected $name;
}

interface Handler2
{
    public function handle($payload);
}

class Service2 extends Base2 implements Handler2
{
    public function handle($payload): bool
    {
        $this->validate($payload);
        return true;
    }

    private function validate($payload): void
    {
    }
}

function helper_2($value) {
    return $value;
}

File: tests/fixtures/edgecase_repo/ruby/module2.rb
require_relative "./module1"

module Example
  class Base2
    def initialize(name)
      @name = name
    end
  end

  class Service2 < Base2
    def handle(payload)
      validate(payload)
      true
    end

    def validate(payload)
      !payload.nil?
    end
  end
end

def helper_2(value)
  value
end
